fish-rust/src/job_group.rs

```diff
@@ -0,0 +1,352 @@
+use self::job_group::pgid_t;
+use crate::common::{assert_send, assert_sync};
+use crate::wchar_ffi::{WCharFromFFI, WCharToFFI};
+use cxx::{CxxWString, UniquePtr};
+use std::num::{NonZeroI32, NonZeroU32};
+use std::sync::atomic::{AtomicBool, AtomicI32, Ordering};
+use std::sync::Mutex;
+use widestring::WideUtfString;
+
+#[cxx::bridge]
+mod job_group {
+    // Not only does cxx bridge not recognize libc::pid_t, it doesn't even recognize i32 as a POD
+    // type! :sadface:
+    struct pgid_t {
+        value: i32,
+    }
+
+    extern "Rust" {
+        #[cxx_name = "job_group_t"]
+        type JobGroup;
+
+        fn wants_job_control(&self) -> bool;
+        fn wants_terminal(&self) -> bool;
+        fn is_foreground(&self) -> bool;
+        fn set_is_foreground(&self, value: bool);
+        #[cxx_name = "get_command"]
+        fn get_command_ffi(&self) -> UniquePtr<CxxWString>;
+        #[cxx_name = "get_job_id"]
+        fn get_job_id_ffi(&self) -> i32;
+        #[cxx_name = "get_cancel_signal"]
+        fn get_cancel_signal_ffi(&self) -> i32;
+        #[cxx_name = "cancel_with_signal"]
+        fn cancel_with_signal_ffi(&self, signal: i32);
+        fn set_pgid(&mut self, pgid: i32);
+        #[cxx_name = "get_pgid"]
+        fn get_pgid_ffi(&self) -> UniquePtr<pgid_t>;
```

Comment:
I would have retunred an `i32` but this is fine too.
If i32 really doesn't work that's surprising to say the least.

Reply:
i32 works but then we need to use a sentinel value like `-1` to indicate `None`.